pydrive2/test/test_util.py

```diff
@@ -1,9 +1,53 @@
+import unittest
 import random
 import re
+import os
+
+from funcy import retry, compose, decorator
+from pydrive2.files import ApiRequestError
 
 newline_pattern = re.compile(r'[\r\n]')
 
 
+class PyDriveTestBase(unittest.TestCase):
+    GDRIVE_USER_CREDENTIALS_DATA = "GDRIVE_USER_CREDENTIALS_DATA"
+    DEFAULT_USER_CREDENTIALS_FILE = "credentials/1.dat"
+    if os.getenv(GDRIVE_USER_CREDENTIALS_DATA):
+        if not os.path.exists(os.path.dirname(DEFAULT_USER_CREDENTIALS_FILE)):
+            os.makedirs(os.path.dirname(DEFAULT_USER_CREDENTIALS_FILE), exist_ok=True)
+        with open(
+                DEFAULT_USER_CREDENTIALS_FILE, "w"
+        ) as credentials_file:
+            credentials_file.write(
+                os.getenv(GDRIVE_USER_CREDENTIALS_DATA)
+            )
+
+class GDriveRetriableError(Exception):
+    pass
+
+@decorator
+def _wrap_pydrive_retriable(call):
```

Comment:
PTAL at the latest cleanup by @Suor in the main repo - he has wrapped it nicely with some nice new features from `fancy`

Reply:
So far, I've left implementation as is (it was changed and simplified since this your comment). The difference with DVC is in usage: in-lined lambda vs decorator.